Sources/NostrSDK/Events/MuteListEvent.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  MuteListEvent.swift
+//
+//
+//  Created by Bryan Montz on 12/15/23.
+//
+
+import Foundation
+
+/// An event that contains various things the user doesn't want to see in their feeds.
+///
+/// See [NIP-51](https://github.com/nostr-protocol/nips/blob/master/51.md#standard-lists).
+public final class MuteListEvent: NostrEvent, HashtagInterpreting, DirectMessageEncrypting {
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+    
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+    
+    init(content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: .muteList, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+    
+    /// The publicly muted public keys (authors).
+    public var pubkeys: [String] {
+        allValues(forTagName: .pubkey) ?? []
+    }
+    
+    /// The publicly muted event ids (threads).
+    public var eventIds: [String] {
+        allValues(forTagName: .event) ?? []
+    }
+    
+    /// The publicly muted hashtags.
+    public var hashtags: [String] {
```

Comment:
I'm surprised that the compiler didn't complain about `hashtags` being declared twice due to this class extending `HashtagInterpreting`.

On a related note, we can update `HashtagInterpreting` and a few other classes to use the `allValues` helper. If you didn't want to add noise and bloat to this PR, we can do that in a separate PR as a small refactor.

Reply:
In a protocol extension, the function in the extension is just the "default" implementation and can be overridden by conformers.

Yep, coming in a separate PR.

Reply:
I'll remove this function's implementation though, since it's just duplicating the one in the protocol.